<?xml version="1.0"?>
<robot name="lujiebot" xmlns:xacro="http://www.ros.org/wiki/xacro">

  <property name="base_length" value="0.68" /> 
  <property name="base_width" value="0.53" /> 
  <property name="laser_height" value="0.275" /> 
  <property name="hokuyo_size" value="0.02" /> 
  <property name="M_PI" value="3.1415926" /> 
  
  
  
  
  <link name="base_footprint">
    <inertial>
      <mass value="0.0001" />
      <origin xyz="0 0 0" />
      <inertia ixx="0.0001" ixy="0.0" ixz="0.0"
          iyy="0.0001" iyz="0.0" 
          izz="0.0001" />
    </inertial>
    <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
        <geometry>
            <box size="0.001 0.001 0.001" />
        </geometry>
    </visual>
  </link>

  <link name="hokuyo_link_front">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="${hokuyo_size} ${hokuyo_size} ${hokuyo_size}"/>
      </geometry>
      <material name="Blue" />
    </visual>
  </link>

  <joint name="hokuyo_joint_front" type="fixed">
    <origin xyz="${base_length/2}  ${-base_width/2} ${laser_height}" rpy="0 0 ${M_PI/2}" />
    <parent link="base_footprint"/>
    <child link="hokuyo_link_front" />
  </joint>


  <link name="hokuyo_link_back">
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="${hokuyo_size} ${hokuyo_size} ${hokuyo_size}"/>
      </geometry>
      <material name="Blue" />
    </visual>
  </link>


  <joint name="hokuyo_joint_back" type="fixed">
    <origin xyz="${-base_length/2}  ${base_width/2} ${laser_height}" rpy="0 0 ${-M_PI/2}" />
    <parent link="base_footprint"/>
    <child link="hokuyo_link_back" />
  </joint>


</robot>